### PR TITLE
Port prog1 and prog2

### PR DESCRIPTION
**Pull request by shaleh**
Add a `both` method to `LispCons` which retuns car and cdr. Also,
prog_ignore

### CODE REVIEW

**Comment by shaleh** [comment]
Comments on the naming of the `both` method? I could not think of a better name short of `car_and_cdr`.

File: rust_src/src/lisp.rs [hunk 1]
```diff
@@ -1137,6 +1137,10 @@ impl LispCons {
         LispObject::from_raw(unsafe { (*self._extract()).cdr })
     }
 
+    pub fn both(self) -> (LispObject, LispObject) {
```
Comment (Wilfred) [inline rust_src/src/lisp.rs]: How about .as_tuple() or .destructure()?

File: rust_src/src/eval.rs [hunk 2]
```diff
@@ -1,91 +1,138 @@
...
 }
 
-/// Macro to format a "wrong argument type" error message.
-macro_rules! wrong_type {
-    ($pred:expr, $arg:expr) => {
-        xsignal!(::remacs_sys::Qwrong_type_argument, LispObject::from_raw($pred), $arg);
-    };
+/// Evaluate BODY sequentially, discarding its value.
+#[no_mangle]
+pub extern "C" fn prog_ignore(body: Lisp_Object) {
```
Comment (Wilfred) [inline rust_src/src/eval.rs]: Is this used elsewhere? I can only see one reference to it, so I'm wondering if we need it.

Comment (shaleh) [reply rust_src/src/eval.rs]: It is used twice in eval.c and once in bytecode.c.

**Comment by shaleh** [comment]
This PR builds on the previous PR. Once that one lands I can rebase this one.

**Pull request merged**